google/cloud/spanner_v1/session.py

```diff
@@ -273,6 +286,7 @@ def read(self, table, columns, keyset, index="", limit=0, column_info=None):
         :rtype: :class:`~google.cloud.spanner_v1.streamed.StreamedResultSet`
         :returns: a result set instance which can be used to consume rows.
         """
+        self._last_use_time = datetime.now()
```

Comment:
Instead of updating the `_last_use_time` attribute here, should we consider updating it in the final layer where the RPC call is actually made? This would likely require changes to the snapshot.py and transaction.py files.
 For reference: https://github.com/googleapis/python-spanner/blob/7df93ca9b11a5cd2fff448587d87d63e29d7a828/google/cloud/spanner_v1/snapshot.py#L180


There are two main reasons for this approach:
1. The session.snapshot method internally calls the Snapshot.read() function. So modifying snapshot file will be a single file change.
2. As far as I understand, it is uncommon for customers to directly call the read method from the session class. Instead, they typically use the snapshot class, as shown in the example below:
```
with database.snapshot() as snapshot:
    results = snapshot.read("SELECT 1")
```

Reply:
That is good point. This implementation actually also misses a lot of other places where a session is used, for example in transaction.py and batch.py.

I actually think that we can move almost all of the updates of _last_use_time into the trace_call(..) function. That function is used for all RPCs that use a session. I'll update the PR along those lines.